manifests/renew.pp

```diff
@@ -76,12 +80,56 @@
   ]).filter | $arg | { $arg =~ NotUndef and $arg != [] }
   $command = join($_command, ' ')
 
-  cron { 'letsencrypt-renew':
-    ensure   => $cron_ensure,
-    command  => $command,
-    user     => 'root',
-    hour     => $cron_hour,
-    minute   => $cron_minute,
-    monthday => $cron_monthday,
+  case $timer_type {
+    'cron': {
+      cron { 'letsencrypt-renew':
+        ensure   => $cron_ensure,
+        command  => $command,
+        user     => 'root',
+        hour     => $cron_hour,
+        minute   => $cron_minute,
+        monthday => $cron_monthday,
+      }
+    }
+    'systemd': {
+      $systemd_ensure = $facts['service_provider'] ? {
+        'systemd' => present,
+        default   => absent,
+      }
+
+      systemd::unit_file { 'letsencrypt.timer':
+        ensure  => $systemd_ensure,
+        content => epp('letsencrypt/letsencrypt.timer.epp'),
+      }
+
+      systemd::unit_file { 'letsencrypt.service':
+        ensure  => $systemd_ensure,
+        content => epp('letsencrypt/letsencrypt.service.epp'),
+      }
+
+      if systemd_ensure {
+        $systemd_timer = 'running'
+      } else {
+        $systemd_timer = $systemd_ensure
+      }
+
+      service { 'letsencrypt.timer':
```

Comment:
with more modern version of our systemd module, you can combine `systemd::unit_file { 'letsencrypt.timer':`, `systemd::unit_file { 'letsencrypt.service':` and the service resource:
https://github.com/voxpupuli/puppet-systemd#timer-units